.github/workflows/build-deploy-pudl.yml

```diff
@@ -33,29 +33,46 @@ jobs:
         uses: actions/checkout@v4
         with:
           ref: ${{ env.BUILD_REF }}
+          fetch-depth: 0
+
+      - name: Skip the build if no changes since the last successful nightly build.
+        run: |
+          CURRENT_COMMIT=$(git rev-parse HEAD)
+          NIGHTLY_COMMIT=$(git rev-parse origin/nightly)
+          if [[ "$CURRENT_COMMIT" == "$NIGHTLY_COMMIT" ]]; then
+            echo "::notice::No changes since last successful nightly build. Skipping."
+            echo "SKIP_BUILD=true" >> $GITHUB_ENV
+            exit 0
+          fi
 
       - name: Set action environment variables
+        if: ${{ env.SKIP_BUILD != 'true' }}
```

Comment:
Seems like there should be a more elegant way to skip everything once we've decided to bail but what is it?  Maybe split into 2 jobs, rather than using steps?

Reply:
Sure seems like it, but I guess what you've done is the best solution for now: https://github.com/orgs/community/discussions/82744

Reply:
Ah okay, if that's as good as it gets, it does work!